### PR TITLE
New function: Get Substance MeSH ID's for given PMID

### PR DESCRIPTION
**Pull request by samuelbunga**
This PR consists of a new function to the pubmed_client module. It takes PubMed ID as an input and returns back a list of substance MeSH ID as an output.

```
from indra.literature.pubmed_client import get_substance_mesh_id
mesh_ids = get_substance_mesh_id('27959613')
```
```
Out[4]: 
['D009570',
 'D010879',
 'D047428',
 'D011725',
 'D011960',
 'D014230',
 'D000077289',
 'D018844',
 'C500026']
```

### CODE REVIEW

**Comment by bgyori** [comment]
@samuelbunga thanks, looks good! But this function should rather be in `indra.literature.pubmed_client`. Could you move it there?

**Comment by samuelbunga** [comment]
> @samuelbunga thanks, looks good! But this function should rather be in `indra.literature.pubmed_client`. Could you move it there?

Sure @bgyori  I will move it. Thanks!

File: indra/databases/pubchem_client.py [hunk 1]
```diff
@@ -125,6 +126,38 @@ def get_mesh_id(pubchem_cid: str) -> Union[str, None]:
...
+
+    Returns
+    -------
+    :
+        Substance MeSH ID's corresponding to the given PubMed paper or
+        if None present or a failed query, an empty list will be returned.
+
+    """
+    url = '%sdb=%s&id=%s&retmode=text&rettype=XML' % (eutils_url, 'pubmed', pubmed_id)
+    res = requests.get(url)
```
Comment (cthoyt) [inline indra/databases/pubchem_client.py:145]: you can use the `params` kwarg in `requests.get` to do all of that formatting for you. just give it a dictionary of the keys to values (don't forget to remove the ?) from the URL

File: indra/databases/pubchem_client.py [hunk 2]
```diff
@@ -125,6 +126,38 @@ def get_mesh_id(pubchem_cid: str) -> Union[str, None]:
...
+    root = ET.fromstring(res.content)
+    nodes = root.findall('.//MedlineCitation/ChemicalList')
+    if len(nodes) == 0:
+        logger.error('Could not retrieve substance MeSH IDs for %s' % pubmed_id)
+        return []
+
+    uid = []
```
Comment (cthoyt) [inline indra/databases/pubchem_client.py:152]: use a list comprehension ;)

File: indra/databases/pubchem_client.py (hunk 1 above)
Comment (bgyori) [reply indra/databases/pubchem_client.py:145]: actually, that call shouldn't be necessary, @samuelbunga there is already a function called `get_full_xml` to get an element tree.

Comment (samuelbunga) [reply indra/databases/pubchem_client.py]: Oh I just saw this function `get_full_xml`
Thanks @bgyori  and @cthoyt

File: indra/databases/pubchem_client.py (hunk 2 above)
Comment (samuelbunga) [reply indra/databases/pubchem_client.py]: Yup !! ;)

File: indra/databases/pubchem_client.py [hunk 3]
```diff
@@ -2,12 +2,13 @@
...
 from typing import List, Union
 from functools import lru_cache
+import xml.etree.ElementTree as ET
 from indra.resources import get_resource_path
 from indra.util import read_unicode_csv
 
 
 pubchem_url = 'https://pubchem.ncbi.nlm.nih.gov/rest/pug'
-
+eutils_url = 'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?'
```
Comment (bgyori) [inline indra/databases/pubchem_client.py:11]: ^ the above changes in this file can be reverted

**Review by bgyori — APPROVED** [review]

**Pull request merged**